### PR TITLE
Correctly add class path entries

### LINKED ISSUE #3592: Hot reload fails with InvalidPathException on Windows
Starting with Quarkus 0.21.0 hot reload failes with InvalidPathException (at least) on Windows.

When a Quarkus application runs in dev mode, i. e. using mvn quarkus:dev, a REST call will trigger a hot reload, if one of the source files has changed before. That worked like a charm up to version 0.20.0. With 0.21.0 the REST response contains an error message:

`<header>
    <h1 class="container">Error restarting Quarkus</h1>
    <div class="exception-message">
        <h2 class="container">java.nio.file.InvalidPathException: Illegal char &lt;:&gt; at index 2: /C:/Users/dw/.m2/repository/io/quarkus/arc/arc/0.21.0/arc-0.21.0.jar</h2>
    </div>
</header>`

As the message complains about the ':' in the file path, I guess, the problem arises on Windows only.

The bug can easily be reproduced:
- Use any of the applications from the guide, e. g getting-started
- Make sure that quarkus.version in pom.xml is 0.21.0
- Build the application and start dev mode (mvn clean package quarkus:dev)
- Make some code change
- Call a REST endpoint (curl http://localhost:8080/hello)

My environment:
- Windows 10 Pro
- AdoptOpenJDK 8 (jdk8u202-b08) and 11 (jdk-11.0.3+7)

### PR DESCRIPTION
**Pull request by dmlloyd**
Fixes #3592.  Windows URI paths are not correctly extracted.

### CODE REVIEW

**Comment by geoand** [comment]
Seems like imports are not sorted

**Comment by geoand** [comment]
If you are using `IntelliJ`, it might happing on commit where the IDE tries to optimize imports before doing the actual commit. It's happened to me plenty of times when I forgot the relevant setting enabled

**Review by cescoffier — APPROVED** [review]

**Review by stuartwdouglas — DISMISSED** [review]
Failures appear related

**Comment by gsmet** [comment]
@dmlloyd any chance you could have another look? It would be nice to have this one in tomorrow's release! Thanks!

**Comment by dmlloyd** [comment]
Updated the comment here with relevant info: https://github.com/quarkusio/quarkus/issues/1673#issuecomment-477205320

File: integration-tests/kogito-maven/src/test/java/io/quarkus/kogito/maven/it/KogitoDevModeIT.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
     @Test
     public void testThatTheKogitoApplicationRuns() throws MavenInvocationException, IOException {
         testDir = initProject("projects/simple-kogito", "projects/project-classic-run-kogito");
-        run();
+        run("-e");
```
Comment (gsmet) [inline integration-tests/kogito-maven/src/test/java/io/quarkus/kogito/maven/it/KogitoDevModeIT.java]: Do we want to keep that?

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 2]
```diff
@@ -430,6 +433,11 @@ private void addToClassPaths(StringBuilder classPathManifest, DevModeContext cla
             throw new RuntimeException(e);
         }
         String path = uri.getRawPath();
+        if (PropertyUtils.isWindows()) {
+            if (path.length() > 2 && Character.isLetter(path.charAt(0)) && path.charAt(1) == ':') {
+                path = "/" + path;
+            }
+        }
```
Comment (gsmet) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: Wondering if we should have an util for that as we have the exact same dark magic code in 2 places? Not a blocker, just a though.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a couple of questions/comments.

File: integration-tests/kogito-maven/src/test/java/io/quarkus/kogito/maven/it/KogitoDevModeIT.java (hunk 1 above)
Comment (dmlloyd) [reply integration-tests/kogito-maven/src/test/java/io/quarkus/kogito/maven/it/KogitoDevModeIT.java]: Yes IMO.  It doesn't do any harm, and you always want to add it when things go wrong.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java (hunk 2 above)
Comment (dmlloyd) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java]: Maybe at some point.  There are probably a lot of things that can be cleaned up in this code.

**Comment by dmlloyd** [comment]
@gsmet, are we good to go?

**Review by gsmet — APPROVED** [review]

**Pull request merged**